simtools/simtel/simtel_runner_ray_tracing.py

```diff
@@ -0,0 +1,256 @@
+
+        Parameters
+        ----------
+        telescopeModel: str
+            Instance of TelescopeModel class.
+        label: str, optional
+            Instance label. Important for output file naming.
+        simtelSourcePath: str (or Path), optional
+            Location of sim_telarray installation. If not given, it will be taken from the
+            config.yml file.
+        filesLocation: str (or Path), optional
+            Parent location of the output files created by this class. If not given, it will be
+            taken from the config.yml file.
+        configData: dict.
+            Dict containing the configurable parameters.
+        configFile: str or Path
+            Path of the yaml file containing the configurable parameters.
+        singleMirrorMode: bool
+            True for single mirror simulations.
+        '''
+        self._logger = logging.getLogger(__name__)
+        self._logger.debug('Init SimtelRunnerRayTracing')
+
+        super().__init__(
+            label=label,
+            simtelSourcePath=simtelSourcePath,
+            filesLocation=filesLocation
+        )
+
+        self.telescopeModel = self._validateTelescopeModel(telescopeModel)
+        self.label = label if label is not None else self.telescopeModel.label
+
+        # File location
+        self._baseDirectory = io.getOutputDirectory(
+            self._filesLocation,
+            self.label,
+            'ray-tracing'
+        )
+        self._baseDirectory.mkdir(parents=True, exist_ok=True)
+
+        self._singleMirrorMode = singleMirrorMode
+
+        # RayTracing - default parameters
+        self._repNumber = 0
+        self.RUNS_PER_SET = 1 if self._singleMirrorMode else 20
+        self.PHOTONS_PER_RUN = 10000
+
+        # Loading configData
+        _configDataIn = gen.collectDataFromYamlOrDict(configFile, configData)
+        _parameterFile = io.getDataFile('parameters', 'simtel-runner-ray-tracing_parameters.yml')
+        _parameters = gen.collectDataFromYamlOrDict(_parameterFile, None)
+        self.config = gen.validateConfigData(_configDataIn, _parameters)
+
+        self._loadRequiredFiles()
+
+    def _loadRequiredFiles(self):
+        '''
+        Which file are required for running depends on the mode.
+        Here we define and write some information into these files. Log files are always requires.
+        '''
+
+        self._corsikaFile = self._simtelSourcePath.joinpath('run9991.corsika.gz')
+
+        # Loop to define and remove existing files.
+        # Files will be named _baseFile = self.__dict__['_' + base + 'File']
+        for baseName in ['stars', 'photons', 'log']:
+            fileName = names.rayTracingFileName(
+                self.telescopeModel.site,
+                self.telescopeModel.name,
+                self.config.sourceDistance,
+                self.config.zenithAngle,
+                self.config.offAxisAngle,
+                self.config.mirrorNumber if self._singleMirrorMode else None,
```

Comment:
Does that mean one cannot run >1 mirror telescope (I like misunderstand the line)

Reply:
No, one can run the full dish or single mirror telescopes. Why? I don't get it.

Reply:
Wait: single mirror == 1 mirror panel? I though we discuss here single vs dual-mirror (dish) telescopes.

I guess we need to have a clear understanding of the language, mirror is really unclear.

Reply:
Yes, I'm talking about mirror panels, not dishes.